app.rb

```diff
@@ -0,0 +1,87 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield self[i]
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield self[i], i
+      i += 1
+    end
+  end
+
+  def my_select
+    new_numbers = []
+    each do |i|
+      new_numbers.push(i) if yield(i)
+    end
+    new_numbers
+  end
+
+  def my_all?
+    condition = true
+    each do |j|
+      condition = false unless yield(j)
+    end
+    condition
+  end
```

Comment:
1. When no block or argument is given should return true when none of the collection members are false or nil or should return false if when one of the collection members is false or nil. For example:
`[1, true, 'hi', []].my_all?` - should return true

2. When a class is passed as an argument should return true if all of the collection is a member of such class::Integer. For example:
`[1, 2, 3].my_all?(Integer)` - should return true

3. When a class is passed as an argument should return true if all of the collection is a member of such class::Numeric. For example:
`[1, 2i, 3.14].my_all?(Numeric)` - should return true

4. When a class is passed as an argument should return false if any of the collection is not a member of such class. For example:
`[1, 'dog', []].my_all?(Integer)` - should return false

5. When a Regex is passed as an argument should return true if all of the collection matches the Regex or should return false if any of the collection does not match the Regex. For example:
`['dog','door','dish'].my_all?(/d/)` - should return true

6. When a pattern other than Regex or a Class is given should return true if all of the collection matches the pattern or should return false if any of the collection does not match the pattern. For example:
`[3, 3, 3].my_all?(3)` - should return true